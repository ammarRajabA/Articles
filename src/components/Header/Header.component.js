import React,{Component} from 'react';

import Menu from '../Menu/Menu.component.js';

import './Header.style.css'

class Header extends Component{
	state={showMenu:false}
	renderMenu=()=>{
		return(
			<div className="menu-container">
	          <span className="menu-button" onClick={()=>this.setState({showMenu:true})}>
	            <i className="fas fa-bars"></i>
	          </span>
	          <Menu show={this.state.showMenu} links={this.props.links} onDismiss={()=>this.setState({showMenu:false})}/>
	        </div>
			)
	}
	render(){
		return (
				<div className="header-container">
					<span className="header-logo-container">
						<img className="logo" src="/logo.png"/>
						<span className="label">Articles</span>
					</span>
					{
						Object.keys(this.props.links).map((link,index)=>
								<span key={index.toString()} className="header-item" onClick={this.props.links[link]}>{link}</span>
							)
					}
					<div className="header-childs">
						{this.renderMenu()}
						{this.props.children}
					</div>
				</div>
			)
	}
}

export default Header;