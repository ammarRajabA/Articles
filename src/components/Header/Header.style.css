.header-container{
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: flex-start;
	background-image: linear-gradient(to right, #EF2A29 , #F05229);
	padding-top: 0.5vw;
	padding-bottom: 1.5vw;
}
.header-container .header-item{
	font-size: 2.5vmin;
	font-family: calibri;
	font-weight: bold;
	color: #F1F1F1;
	margin-right: 1em;
	margin-left: 1em;
	cursor: pointer;
	transition: 0.3s;
}
.header-container .header-item:hover{
	color: #FFFFFF;
}
.header-container .header-logo-container{
	display: flex;
	flex-direction: row;
	align-items: center;
}
.header-logo-container .logo{
	width: 8vmin;
	margin-left: 4vmin;
	margin-right: 2vmin;
}
.header-logo-container .label{
	font-size: 4vmin;
	margin-right: 4vmin;
	color: white;
}
.header-container .header-childs{
	margin-left: auto;
	margin-right: 10%;
}

.header-container .menu-container{
	display: none;
}

@media (min-width: 320px) and (max-width: 767px) {
	.header-container .header-item{
		display: none;
	}
	.header-container .menu-button{
		padding: 10px;
		font-size: 14px;
		cursor: pointer;
	}
	.header-container .menu-container{
		position: relative;
		display: block;
		top:5px;
		right: -10px;
	}
}