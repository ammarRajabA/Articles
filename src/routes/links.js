import history from './history';

export default {
	"Home":()=>history.push('/'),
	"Articles":()=>history.push('/articles'),
	"New Article":()=>history.push('/new-article'),
	"About":()=>history.push('/about')
}