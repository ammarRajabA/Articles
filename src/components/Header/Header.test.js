import React from 'react'
import { shallow ,mount } from 'enzyme';

import Header from './Header.component';
import links from '../../routes/links'

describe('Header component',function () {
	it('match snapshot',function () {
		const component = mount(<Header links={links}/>);
		expect(component).toMatchSnapshot();
	})
})