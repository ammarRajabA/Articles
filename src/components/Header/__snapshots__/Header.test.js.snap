// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component match snapshot 1`] = `
<Header
  links={
    Object {
      "About": [Function],
      "Articles": [Function],
      "Home": [Function],
      "New Article": [Function],
    }
  }
>
  <div
    className="header-container"
  >
    <span
      className="header-logo-container"
    >
      <img
        className="logo"
        src="/logo.png"
      />
      <span
        className="label"
      >
        Articles
      </span>
    </span>
    <span
      className="header-item"
      key="0"
      onClick={[Function]}
    >
      Home
    </span>
    <span
      className="header-item"
      key="1"
      onClick={[Function]}
    >
      Articles
    </span>
    <span
      className="header-item"
      key="2"
      onClick={[Function]}
    >
      New Article
    </span>
    <span
      className="header-item"
      key="3"
      onClick={[Function]}
    >
      About
    </span>
    <div
      className="header-childs"
    >
      <div
        className="menu-container"
      >
        <span
          className="menu-button"
          onClick={[Function]}
        >
          <i
            className="fas fa-bars"
          />
        </span>
        <Menu
          links={
            Object {
              "About": [Function],
              "Articles": [Function],
              "Home": [Function],
              "New Article": [Function],
            }
          }
          onDismiss={[Function]}
          show={false}
        >
          <span />
        </Menu>
      </div>
    </div>
  </div>
</Header>
`;
